cheatsheets/Clickjacking_Defense_Cheat_Sheet.md

```diff
@@ -311,3 +312,125 @@ Activate [designMode](https://developer.mozilla.org/en-US/docs/Web/API/Document/
 ```javascript
 document.designMode = "on";
 ```
+ # Double Clickjacking 
+
+## Introduction 
+
+Double Clickjacking is an advanced form of Clickjacking that exploits user interactions by requiring two consecutive clicks to execute a malicious action. Attackers use this technique to bypass security mechanisms that rely on single-click protections, making it more difficult to detect and prevent. Unlike traditional Clickjacking, which usually relies on a single user action, Double Clickjacking introduces an additional interaction to increase the success rate of the attack and evade common security measures.
+
+## Attack Scenario
+
+1. The attacker loads a malicious webpage that contains a transparent iframe overlaying a legitimate website.
+
+2. The user is tricked into clicking an element, such as a button or link, thinking it belongs to the attacker's site.
+
+3. The first click moves the transparent iframe into position over a critical UI element of the target website.
+
+4. The second click executes the action, such as transferring funds, changing security settings, or posting content without user consent.
+
+## Mitigation Strategies
+
+To defend against Double Clickjacking, implement the following strategies:
+
+### 1. Frame Busting
+
+Prevent your site from being embedded within an iframe by using frame-busting techniques, such as:
+
+<script>
+if (self !== top) {
+    top.location = self.location;
+}
+</script>
+
+#### Considerations:
+
+Frame-busting scripts help prevent iframes but can be bypassed in some cases using JavaScript modifications.
+
+Attackers may use techniques like dynamically injecting iframes post-load to evade frame-busting mechanisms.
+
+### 2. X-Frame-Options Header
+
+Use the X-Frame-Options HTTP header to restrict iframe embedding:
+
+X-Frame-Options: DENY
+
+Alternatively, allow only trusted domains:
+
+X-Frame-Options: SAMEORIGIN
+
+#### Limitations:
+
+Some browsers do not support X-Frame-Options.
+
+Attackers can still use UI redressing techniques to manipulate users into clicking unintended elements.
+
+### 3. Content Security Policy (CSP) Frame-Ancestors
+
+Use CSP to control which domains can embed your site:
+
+Content-Security-Policy: frame-ancestors 'self' https://trusted.example.com;
+
+#### Limitations:
+
+CSP-based protections only work when enforced properly and do not protect against all forms of Clickjacking.
+
+Attackers can still use overlays or timing-based click manipulation to deceive users.
+
+### 4. Double-Click Confirmation
+
+Implement a double-click confirmation mechanism for critical actions to prevent unintended interactions.
+#### Example:
+
+<button onclick="confirmAction()">Submit</button>
+<script>
+function confirmAction() {
+    if (!this.clickedOnce) {
+        this.clickedOnce = true;
+        alert('Click again to confirm action.');
+    } else {
+        // Proceed with the action
+    }
+}
```

Comment:
@jmanico @mackowski @szh - JavaScript is not my forte. In fact, I only learn enough of it to create some very simple PoC exploits, so I am not qualified to review this. If I were, I would have submitted a PR myself.